cumulus/zombienet/zombienet-sdk/tests/zombie_ci/pov_recovery.rs

```diff
@@ -0,0 +1,226 @@
+
+const PARA_ID: u32 = 2000;
+
+// This tests makes sure that the recovering nodes are able to see all relay-chain
+// notifications containing the candidates to recover.
+#[tokio::test(flavor = "multi_thread")]
+async fn pov_recovery() -> Result<(), anyhow::Error> {
+	let _ = env_logger::try_init_from_env(
+		env_logger::Env::default().filter_or(env_logger::DEFAULT_FILTER_ENV, "info"),
+	);
+
+	log::info!("Spawning network with relay chain only");
+	let mut network = initialize_network().await?;
+
+	let validator_3 = network.get_node("validator-3")?;
+
+	log::info!("Waiting 20 blocks to register parachain");
+	// Wait 20 blocks and register parachain. This part is important for pov-recovery.
+	assert!(validator_3
+		.wait_metric_with_timeout(BEST_BLOCK_METRIC, |b| b >= 20.0, 250u64)
+		.await
+		.is_ok());
+
+	log::info!("Registering parachain para_id = {PARA_ID}");
+	network.register_parachain(PARA_ID).await?;
+
+	let relay_ferdie = network.get_node("ferdie")?;
+	let relay_client: OnlineClient<PolkadotConfig> = relay_ferdie.wait_client().await?;
+
+	log::info!("Ensuring parachain is registered");
+	assert_para_throughput(
+		&relay_client,
+		20,
+		[(ParaId::from(PARA_ID), 2..40)].into_iter().collect(),
+	)
+	.await?;
+
+	log::info!("Checking block production");
+	for (name, timeout_secs) in [
+		("bob", 600u64),
+		("alice", 600u64),
+		("charlie", 600u64),
+		("one", 800u64),
+		("two", 800u64),
+		// Re-enable once we upgraded from smoldot 0.11.0 and https://github.com/paritytech/polkadot-sdk/pull/1631 is merged
+		// ("three", 800u64),
+		("eve", 800u64),
+	] {
+		assert!(network
+			.get_node(name)?
+			.wait_metric_with_timeout(BEST_BLOCK_METRIC, |b| b >= 20.0, timeout_secs)
+			.await
+			.is_ok());
+	}
+
+	log::info!("Ensuring blocks are imported using PoV recovery");
+
+	// Wait (up to 10 seconds) until pattern occurs 20 times
+	let options = LogLineCountOptions {
+		predicate: Arc::new(|n| n == 20),
+		timeout: Duration::from_secs(10),
+		wait_until_timeout_elapses: false,
+	};
+
+	for name in ["one", "two", "three", "eve", "charlie", "alice"] {
+		let result = network
+			.get_node(name)?
+			.wait_log_line_count_with_timeout(
+				"Importing blocks retrieved using pov_recovery",
+				false,
+				options.clone(),
+			)
+			.await?;
+
+		assert!(result.success());
+	}
+
+	Ok(())
+}
+
+async fn initialize_network() -> Result<Network<LocalFileSystem>, anyhow::Error> {
+	let images = zombienet_sdk::environment::get_images_from_env();
+	log::info!("Using images: {images:?}");
+
+	// Network setup:
+	// - relaychain nodes:
+	// 	 - ferdie
+	// 	   - full node
+	// 	 - validator[0-13]
+	// 	   - validator
+	// 	   - synchronize only with ferdie
+	// - parachain nodes
+	//   - bob
+	//     - collator which is the only one producing blocks
+	//   - alice
+	//     - collator which doesn't produce blocks
+	//     - will need to recover the pov blocks through availability recovery
+	//   - charlie
+	//     - full node
```

Comment:
The old toml file had a comment wrt charlie recovering the pov blocks, is this still true after migration?

>   # alice and charlie will need to recover the pov blocks through availability recovery

Reply:
oh, good catch! yes this is still true. I will fix the comment for `charlie`.